src/main/java/org/cryptacular/V1CiphertextHeader.java

```diff
@@ -0,0 +1,152 @@
+/* See LICENSE for licensing and NOTICE for copyright. */
+package org.cryptacular;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import org.cryptacular.util.ByteUtil;
+
+/**
+ * Cleartext header prepended to ciphertext providing data required for decryption.
+ *
+ * <p>Data format:</p>
+ *
+ * <pre>
+     +-----+----------+-------+------------+---------+
+     | Len | NonceLen | Nonce | KeyNameLen | KeyName |
+     +-----+----------+-------+------------+---------+
+ * </pre>
+ *
+ * <p>Where fields are defined as follows:</p>
+ *
+ * <ul>
+ *   <li>Len - Total header length in bytes (4-byte integer)</li>
+ *   <li>NonceLen - Nonce length in bytes (4-byte integer)</li>
+ *   <li>Nonce - Nonce bytes (variable length)</li>
+ *   <li>KeyNameLen (OPTIONAL) - Key name length in bytes (4-byte integer)</li>
+ *   <li>KeyName (OPTIONAL) - Key name encoded as bytes in platform-specific encoding (variable length)</li>
+ * </ul>
+ *
+ * <p>The last two fields are optional and provide support for multiple keys at the encryption provider. A common case
+ * for multiple keys is key rotation; by tagging encrypted data with a key name, an old key may be retrieved by name to
+ * decrypt outstanding data which will be subsequently re-encrypted with a new key.</p>
+ *
+ * @author  Middleware Services
+ */
+public class V1CiphertextHeader implements CiphertextHeader
```

Comment:
Why the name change to put V1 at the beginning?

Reply:
I took advantage of a breaking API change to rename components to be more consistent with Java naming conventions.

Reply:
What makes this more consistent?

Reply:
Convention is for concrete classes to end with the interface name.